implementations/rust/ockam/ockam_command/src/vault/delete.rs

```diff
@@ -40,19 +57,64 @@ async fn run_impl(
     opts: CommandGlobalOpts,
     cmd: DeleteCommand,
 ) -> miette::Result<()> {
-    let DeleteCommand { name, yes } = cmd;
-    if opts
-        .terminal
-        .confirmed_with_flag_or_prompt(yes, "Are you sure you want to delete this vault?")?
-    {
-        opts.state.vaults.get(&name)?;
-        opts.state.vaults.delete(&name)?;
-        opts.terminal
+    DeleteTui::run(opts, cmd).await
+}
+
+#[ockam_core::async_trait]
+impl DeleteCommandTui for DeleteTui {
+    const ITEM_NAME: &'static str = "vaults";
+    fn cmd_arg_item_name(&self) -> Option<&str> {
+        self.cmd.name.as_deref()
+    }
+
+    fn cmd_arg_delete_all(&self) -> bool {
+        self.cmd.all
+    }
+
+    fn cmd_arg_confirm_deletion(&self) -> bool {
+        self.cmd.yes
+    }
+
+    fn terminal(&self) -> Terminal<TerminalStream<Term>> {
+        self.opts.terminal.clone()
+    }
+
+    async fn get_arg_item_name_or_default(&self) -> miette::Result<String> {
+        Ok(self
+            .cmd
+            .name
+            .clone()
+            .unwrap_or(self.opts.state.vaults.default()?.name().to_string()))
+    }
+
+    async fn list_items_names(&self) -> miette::Result<Vec<String>> {
+        Ok(self.opts.state.vaults.list_items_names()?)
+    }
+
+    async fn delete_single(&self, item_name: &str) -> miette::Result<()> {
+        self.opts.state.vaults.delete(item_name)?;
+        self.terminal()
             .stdout()
-            .plain(fmt_ok!("Vault with name '{name}' has been deleted"))
-            .machine(&name)
-            .json(serde_json::json!({ "name": &name }))
+            .plain(fmt_ok!("Vault with name '{item_name}' has been deleted"))
+            .machine(item_name)
+            .json(serde_json::json!({ "name": &item_name }))
             .write_line()?;
+
+        Ok(())
+    }
+
+    async fn delete_multiple(&self, selected_items_names: Vec<String>) -> miette::Result<()> {
+        let plain = selected_items_names
+            .iter()
+            .map(|name| {
+                if self.opts.state.vaults.delete(name).is_ok() {
+                    Ok(fmt_ok!("Vault '{name}' deleted\n"))
+                } else {
+                    Ok(fmt_warn!("Failed to delete vault '{name}'\n"))
+                }
```

Comment:
Can you return here the string directly instead of wrapping it in a Result? Not sure if the compiler will complain.

Reply:
Yeah sure, no compiler complaining. My bad, was not needed!